src/transformers/models/beit3/modeling_beit3.py

```diff
@@ -0,0 +1,1460 @@
+            The scaled dot product scores between `text_embeds` and `image_embeds`. This represents the text-image
+            similarity scores.
+        text_embeds(`torch.FloatTensor` of shape `(batch_size, output_dim`):
+            The text embeddings obtained by applying the projection layer to the pooled output of [`Beit3Model`].
+        image_embeds(`torch.FloatTensor` of shape `(batch_size, output_dim`):
+            The image embeddings obtained by applying the projection layer to the pooled output of [`Beit3Model`].
+        text_model_output(`BaseModelOutputWithPooling`):
+            The output of the [`Beit3Model`].
+        vision_model_output(`BaseModelOutputWithPooling`):
+            The output of the [`Beit3Model`].
+    """
+
+    loss: Optional[torch.FloatTensor] = None
+    logits_per_image: torch.FloatTensor = None
+    logits_per_text: torch.FloatTensor = None
+    text_embeds: torch.FloatTensor = None
+    image_embeds: torch.FloatTensor = None
+    text_model_output: BaseModelOutputWithPooling = None
+    vision_model_output: BaseModelOutputWithPooling = None
+
+    def to_tuple(self) -> Tuple[Any]:
+        return tuple(
+            self[k] if k not in ["text_model_output", "vision_model_output"] else getattr(self, k).to_tuple()
+            for k in self.keys()
+        )
+
+
+class Beit3MLP(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+        in_features = config.hidden_size * 4
+        hidden_features = config.hidden_size * 2
+
+        self.norm1 = nn.LayerNorm(in_features, eps=config.layer_norm_eps)
+
+        self.dense1 = nn.Linear(in_features, hidden_features)
+        self.norm2 = nn.LayerNorm(hidden_features, eps=config.layer_norm_eps)
+        self.act = nn.GELU()
+        self.dense2 = nn.Linear(hidden_features, config.num_labels)
+
+    def forward(self, hidden_states: torch.Tensor):
+        hidden_states = self.norm1(hidden_states)
+        hidden_states = self.dense1(hidden_states)
+        hidden_states = self.norm2(hidden_states)
+        hidden_states = self.act(hidden_states)
+        return self.dense2(hidden_states)
+
+
+class Beit3MultiwayFeedForwardNetwork(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+        self.text = Beit3FeedForwardNetwork(config)
+        self.image = Beit3FeedForwardNetwork(config)
+
+    def forward(self, hidden_states: torch.Tensor, split_position: int = -1):
+        if split_position == -1:
+            return self.image(hidden_states)
+        if split_position == 0:
+            return self.text(hidden_states)
+        image_hidden, text_hidden = torch.split(
+            hidden_states,
+            [split_position, hidden_states.size(1) - split_position],
+            dim=1,
+        )
+        image_out, text_out = self.image(image_hidden), self.text(text_hidden)
+        return torch.cat([image_out, text_out], dim=1)
+
+
+class Beit3AttentionLinear(nn.Module):
```

Comment:
I would use a clearer name here - the class doesn't do attention, it does linear projections 

```suggestion
class Beit3AttentionLinearProjection(nn.Module):
```

Reply:
Done